pyas2lib/as2.py

```diff
@@ -985,7 +985,13 @@ def detect_mdn(self):
                 message_id = mdn.get("Original-Message-ID").strip("<>")
                 message_recipient = None
                 if "Original-Recipient" in mdn:
-                    message_recipient = mdn["Original-Recipient"].split(";")[1].strip()
+                    if ";" in mdn["Original-Recipient"]:
```

Comment:
Can we make a util method to extract the recipient, and also can we try using regex for this - `[a-z0-9]+;(.+)|(.+)`

Reply:
Missed your comments, but I did separate into separate function, but kept the split/strip instead of regex and return the address type as well as the address instead of removing the address_type. Not sure if this could become relevant eventually. 
Either way, have also added tests for the "Final Recipient" fallback, which was missing.